### PR TITLE
Candidate for 4.0.19

### PR DESCRIPTION
**Pull request by kushti**
Candidate for 4.0.19. It contains:

* fix for #1547
* fix for #1545 

Block propagation is tested on the testnet

### CODE REVIEW

**Review by pragmaxim — APPROVED** [review]
Looks good besides the failing test :+1:

**Comment by kushti** [comment]
> Looks good besides the failing test +1

Fixed the test

File: src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala [hunk 1]
```diff
@@ -32,7 +32,7 @@ class HistoryStorage(indexStore: LDBKVStore, objectsStore: LDBKVStore, config: C
     .build[ByteArrayWrapper, Array[Byte]]
 
   def modifierBytesById(id: ModifierId): Option[Array[Byte]] =
-    objectsStore.get(idToBytes(id))
+    objectsStore.get(idToBytes(id)).map(_.tail)
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala:35]: @kushti please leave a comment. What this means?

Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/history/storage/HistoryStorage.scala:35]: @aslesarenko fix for https://github.com/ergoplatform/ergo/issues/1547 , will add in 4.0.20

**Pull request merged**